mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/utils/FlagsUpdateStageResult.java

```diff
@@ -46,8 +46,8 @@ public static FlagsUpdateStageResult none() {
 
     @VisibleForTesting
     FlagsUpdateStageResult(List<MessageUid> failed, List<UpdatedFlags> succeeded) {
-        this.failed = failed;
-        this.succeeded = succeeded;
+        this.failed = ImmutableList.copyOf(failed);
```

Comment:
I would rather have changed the argument type to ImmutableList, this is private you could change them anytime if you want and it will avoid this copyOf